### PR TITLE
add recipe for DB_File Perl module

### PR DESCRIPTION
**Pull request by rpetit3**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [X] Source is from official source.
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

This PR adds a recipe for [DB_File](https://github.com/pmqs/DB_File) ([cpan](https://metacpan.org/pod/DB_File)), which allows Perl programs to use Berkeley DB. The only dependencies are a C compiler and Berkeley DB (libdb).

I used the recipe from [perl-xml-parser](https://github.com/bioconda/bioconda-recipes/tree/master/recipes/perl-xml-parser) as a template. Please let me know if any changes are needed.

Thank you!

### CODE REVIEW

**Comment by rpetit3** [comment]
I've created an issue (https://github.com/pmqs/DB_File/issues/8) to see if the developer can provide an explicit LICENSE file. Otherwise I think I'll have to add it with both the Artistic-1.0-Perl OR GPL-1.0-or-later licenses

**Comment by rpetit3** [comment]
@conda-forge/help-perl ready for review!

File: recipes/perl-db_file/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  sha256: d9ffe2a25be6cdfee7a34d64235cdc899e99ba8b3fb8de8a9e7f4af20e4ca960
+
+build:
+  number: 0
+  skip: true  # [win]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - libdb
```
Comment (chrisburr) [inline recipes/perl-db_file/meta.yaml:19]:
```suggestion
```

Comment (rpetit3) [reply recipes/perl-db_file/meta.yaml]: Removed it! Thanks for the suggestion

**Comment by jakirkham** [comment]
@mbargull would you or someone you know have time to look at this? 🙂

**Comment by rpetit3** [comment]
Thank you!

**Pull request merged**